app/javascript/components/game_map.js

```diff
@@ -0,0 +1,63 @@
+import { Axis } from './axis'
+
+class GameMap {
+  constructor(selector) {
+    this._selector = selector
+  }
+
+  addMap() {
+    // Add component html to DOM
+    this._element.innerHTML = this._html()
+
+    // Get the canvas context for the map layer and draw the map image
+    this._drawMapImage(
+      document.getElementById('map-component__map-layer').getContext('2d')
+    )
+
+    this._drawHexGrid(
+      document.getElementById('map-component__grid-layer').getContext('2d')
+    )
+  }
+
+  _axes() {
+    var q = new Axis(JSON.parse(this._element.dataset.q))
+    var r = new Axis(JSON.parse(this._element.dataset.r))
+    var s = new Axis(JSON.parse(this._element.dataset.s))
+
+    return { q, r, s }
```

Comment:
Instead of returning hardcoded names, can this create the variable names based on the `name` property?

Reply:
This now returns an array, which can be iterated over. This makes it more flexible and the names shouldn't matter for anything except the UI.